pytorch_analysis.ipynb

```diff
@@ -393,9 +404,318 @@
+      "text/plain": [
+       "<Figure size 504x360 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "# assemble into dataframe\n",
+    "df_list = []\n",
+    "for j in inter_method_sims:\n",
+    "    df_list.append([j, 'inter', 'signal'])\n",
+    "for j in intra_method_sims_sk:\n",
+    "    df_list.append([j, 'intra_sk', 'signal'])\n",
+    "for j in intra_method_sims_torch:\n",
+    "    df_list.append([j, 'intra_torch', 'signal'])\n",
+    "for j in inter_method_sims_shuffled:\n",
+    "    df_list.append([j, 'inter', 'shuffled'])\n",
+    "for j in intra_method_sims_shuffled_sk:\n",
+    "    df_list.append([j, 'intra_sk', 'shuffled'])\n",
+    "for j in intra_method_sims_shuffled_torch:\n",
+    "    df_list.append([j, 'intra_torch', 'shuffled'])\n",
+    "\n",
+    "jaccard_df = pd.DataFrame(df_list, columns=['jaccard', 'comparison', 'signal'])\n",
+    "sns.set(rc={'figure.figsize': (7, 5)})\n",
+    "sns.barplot(data=jaccard_df, x='signal', y='jaccard', hue='comparison')\n",
+    "plt.title('Jaccard similarity of nonzero coefficients within/between methods')\n",
+    "plt.ylabel('Jaccard similarity')\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### **Key to the above comparisons:**\n",
+    "\n",
+    "* \"Inter-method\" comparison: compare sklearn vs. torch coefficients for the same random seeds/CV folds\n",
+    "* \"Intra-method\" comparison: compare sklearn vs. sklearn (or torch vs. torch) coefficients for the same random seeds across CV folds\n",
+    "* signal vs. shuffled: shuffled scrambles expression between genes (should be no predictive signal)\n",
+    "\n",
+    "### **Metric:**\n",
+    "\n",
+    "Jaccard similarity\n",
+    "\n",
+    "$$J(A, B) = \\frac{|A \\cap B|}{|A \\cup B|}$$\n",
+    "\n",
+    "A and B here are the sets of genes with nonzero coefficients, so this just looks at binary overlap since the scale of the coefficients doesn't matter.\n",
+    "\n",
+    "### **Takeaways:**\n",
+    "\n",
+    "Coefficients are not very stable at all between methods. Despite having similar predictive performance and similar sparsity, the nonzero coefficients found by PyTorch and sklearn have almost no overlap.\n",
```

Comment:
Nice walk through of the results throughout the notebook and a good summary at the end.